Add binary signing example

clehner opened the pull request:
This adds an example of using a Verifiable Credential to assert a content hash and file size.
A shell script uses DIDKit to do signing and verifying, with some AWK for parsing and verifying the claims.
This uses the `didkit to-rdf-urdna2015` subcommand, so includes commits from #172.

sbihel reviewed (approved):
I haven't looked in every detail but it looks good. I had a couple thoughts:
- the code currently in the `main()` should be moved to a function so we can nicely print errors without panicking; and
- maybe we should be careful with AWK scripts as I believe there are significant differences between platforms (but I guess the same could be said about Bash scripts...).

wyc reviewed (changes requested):
I think this is my fault for taking so long to review this PR....my apologies, but it seems to not work with the latest version of `ssi`'s `main` branch:

   Compiling didkit v0.2.1 (didkit/lib)
error[E0061]: this function takes 4 arguments but 3 arguments were supplied
   --> lib/src/lib.rs:136:13
    |
136 |             LinkedDataProofs::sign(document, &options, &jwk).await?
    |             ^^^^^^^^^^^^^^^^^^^^^^ --------  --------  ---- supplied 3 arguments
    |             |
    |             expected 4 arguments
    |
note: associated function defined here
   --> ssi/src/ldp.rs:216:18
    |
216 |     pub async fn sign(
    |                  ^^^^

error[E0061]: this function takes 4 arguments but 3 arguments were supplied
   --> lib/src/ssh_agent.rs:310:16
    |
310 |     let prep = LinkedDataProofs::prepare(document, &options, &jwk).await?;
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^ --------  --------  ---- supplied 3 arguments
    |                |
    |                expected 4 arguments
    |

clehner commented:
@wyc

Rebased, so those errors should no longer occur here.

@sbihel

> maybe we should be careful with AWK scripts as I believe there are significant differences between platforms (but I guess the same could be said about Bash scripts...).

I hope that sticking to POSIX AWK and POSIX Shell helps with this. I tested the use of AWK here with [Gawk (GNU awk)](https://www.gnu.org/software/gawk/), [Mawk](https://invisible-island.net/mawk/), [original-awk](https://tracker.debian.org/pkg/original-awk), and [busybox awk](https://busybox.net/downloads/BusyBox.html#awk). I found one discrepancy, which is the use of interval expressions (e.g. `{1,10}`) in pattern matching. `original-awk` did not support this, although POSIX AWK and the other three implementations do; so I removed use of that feature in b539e63.
Also, `original-awk` treats bytes as characters, while N-Quads is in UTF-8, so there might be a problem there if the input has non-ASCII characters (which the example VC here does not).

wyc reviewed (approved):
Your changes did the trick. Thanks!

wyc commented:
@clehner when you get a chance, would love to see conflicts resolved from the rdf subcommand so we can merge this.

clehner commented:
Rebased and squashed.
